src/sst.rs

```diff
@@ -115,14 +116,15 @@ impl SsTableFormat {
 
     fn decode_index(
         &self,
-        index_bytes: Bytes,
+        bytes: Bytes,
         compression_codec: Option<CompressionCodec>,
     ) -> Result<SsTableIndexOwned, SlateDBError> {
-        let index_bytes = match compression_codec {
+        let index_bytes = self.validate_checksum(bytes)?;
+        let decompressed_bytes = match compression_codec {
             Some(c) => Self::decompress(index_bytes, c)?,
             None => index_bytes,
         };
-        Ok(SsTableIndexOwned::new(index_bytes)?)
+        Ok(SsTableIndexOwned::new(decompressed_bytes)?)
```

Comment:
I'm debating whether it makes sense for `SsTableIndex` to put the checksum into the flatbuffer definition instead of manually manipulating the bytes. otherwise it can be a bit misleading since you can't actually read the raw bytes of the index directly using the protobuf definition if you skip this method.

Reply:
Oh, that's an interesting point... I think you're suggesting:

```
table SsTableInfo {
    // ...
    index_checksum: uint32,
    // ...
}
```

Is that right? I guess we don't do this for the filter/block because neither has a Flatbuffer definition, so it doesn't matter much either way. But since we have a `table SsTableIndex` in the .fbs, it gets a little muddled here. 🤔

Reply:
Another, perhaps related, thought: do we want the checksum _inside_ the compressed data rather than outside? I'm thinking of guarding against the case where an index is compressed with one codec and decompressed with another. In that case, checksum-outside would pass the check, but checksum-inside would fail (which is more correct, IMO).

Reply:
I initially wanted to do that, but then realized that only the index block has a Flatbuffer definition, and unsure about the benefit of storing the raw data (including checksum) in flatbuffer, hence decided to go with a unify manual approach just like filter and block.

Reply:
After some thought, I'm inclined to leave it as-is. My thinking is:

1. Keeping a single style is nice.
2. We already can't pass index_bytes directly into `SsTableIndexOwned` because of compression.
3. The `SsTableIndex` and `SsTableIndexOwned` creation is outside of flatbuffer logic. That is, `sst.fbs` objects don't directly reference `SsTableIndex`, rather we have to read the bytes using the `index_offset`/`index_length` and create a new `SsTableIndex` ourselves.

Reply:
@agavra Ok with you if we merge as-is? Or do you feel there are strong arguments for the alternative?

Reply:
🫡 yeah, I was genuinely torn between which would be better. I think the arguments for keeping it as is makes sense.